catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java

```diff
@@ -161,21 +163,108 @@ public void testToPrimitiveType() {
+                com.datastrato.gravitino.rel.types.Types.IntegerType.get(),
+                false));
+    Assertions.assertTrue(mapTypeNotNull instanceof Types.MapType);
+    Assertions.assertTrue(mapTypeNotNull.asMapType().keyType() instanceof Types.StringType);
+    Assertions.assertTrue(mapTypeNotNull.asMapType().valueType() instanceof Types.IntegerType);
+    Assertions.assertTrue(mapTypeNotNull.asMapType().isValueRequired());
+
+    Type structTypeNullable =
+        ConvertUtil.toIcebergType(
+            com.datastrato.gravitino.rel.types.Types.StructType.of(
+                com.datastrato.gravitino.rel.types.Types.StructType.Field.nullableField(
+                    "col1",
+                    com.datastrato.gravitino.rel.types.Types.MapType.of(
+                        com.datastrato.gravitino.rel.types.Types.StringType.get(),
+                        com.datastrato.gravitino.rel.types.Types.IntegerType.get(),
+                        true)),
+                com.datastrato.gravitino.rel.types.Types.StructType.Field.nullableField(
+                    "col2",
+                    com.datastrato.gravitino.rel.types.Types.ListType.of(
+                        com.datastrato.gravitino.rel.types.Types.FloatType.get(), true))));
+
+    Assertions.assertTrue(structTypeNullable instanceof Types.StructType);
+    Assertions.assertTrue(
+        structTypeNullable.asStructType().fields().get(0).type() instanceof Types.MapType);
+    Assertions.assertTrue(
+        structTypeNullable.asStructType().fields().get(0).type().asMapType().isValueOptional());
+    Assertions.assertTrue(
+        structTypeNullable.asStructType().fields().get(1).type() instanceof Types.ListType);
+    Assertions.assertTrue(
+        structTypeNullable.asStructType().fields().get(1).type().asListType().isElementOptional());
+
+    Type structTypeNotNull =
+        ConvertUtil.toIcebergType(
+            com.datastrato.gravitino.rel.types.Types.StructType.of(
+                com.datastrato.gravitino.rel.types.Types.StructType.Field.notNullField(
+                    "col1",
+                    com.datastrato.gravitino.rel.types.Types.MapType.of(
+                        com.datastrato.gravitino.rel.types.Types.StringType.get(),
+                        com.datastrato.gravitino.rel.types.Types.IntegerType.get(),
+                        false)),
+                com.datastrato.gravitino.rel.types.Types.StructType.Field.notNullField(
+                    "col2",
+                    com.datastrato.gravitino.rel.types.Types.ListType.of(
+                        com.datastrato.gravitino.rel.types.Types.FloatType.get(), false))));
+    Assertions.assertTrue(structTypeNotNull instanceof Types.StructType);
+    Assertions.assertTrue(
+        structTypeNotNull.asStructType().fields().get(0).type() instanceof Types.MapType);
+    Assertions.assertTrue(
+        structTypeNotNull.asStructType().fields().get(0).type().asMapType().isValueRequired());
```

Comment:
for struct type, we should check struct filed is nullable not struct field type

Reply:
added a check:
```
    structTypeNullable.asStructType().fields().forEach(f -> Assertions.assertTrue(f.isOptional()));

    structTypeNullable.asStructType().fields().forEach(f -> Assertions.assertTrue(f.isRequired()));
```